### PR TITLE
{chem}[foss/2018b,intel/2018b] ASE v3.18.0, GPAW v19.8.1 w/ Python 3.6.6

### PR DESCRIPTION
**Pull request by schiotz**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by schiotz** [comment]
This is the latest version of GPAW (version 19.8.1).  It requires the latest ASE, which causes the Travis CI test to fail because two different ASE versions appear in the same toolchain.  Not much I can do about this, but even if this PR cannot be merged it is still useful to have it available for installation with `eb --from-pr=8751`.

CC: @OleHolmNielsen @jjmortensen

**Comment by schiotz** [comment]
I have added an ASE version suffix, this will allow versions of GPAW with different ASE versions to coexist.

**Comment by schiotz** [comment]
Test report by @schiotz
**SUCCESS**
Build succeeded for 0 out of 0 (4 easyconfigs in this PR)
slid.fysik.dtu.dk - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 2.7.5
See https://gist.github.com/eb4c638bffdb7e1d97f3034c820544cd for a full test report.

**Comment by schiotz** [comment]
Test report by @schiotz
**SUCCESS**
Build succeeded for 2 out of 2 (4 easyconfigs in this PR)
svol.fysik.dtu.dk - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/2e92140984b4d8cbeaa71c682c57a0cd for a full test report.

File: easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-customize-intel.patch [hunk 1]
```diff
@@ -0,0 +1,96 @@
+--- /dev/null	2019-08-01 21:28:47.844000266 +0200
```
Comment (boegel) [inline easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-customize-intel.patch]: @schiotz Can you add some comments above this line to clarify what this patch does, and why it is needed?

Also mention the author (or upstream reference).

Don't forget to change the checksum for this patch accordingly...

**Comment by schiotz** [comment]
@boegel Added comment to configuration patch, as requested.  I also added it in #8768 as has to be the same file.

**Comment by schiotz** [comment]
@boegel I believe this one should be ready to merge.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, AMD EPYC 7601 32-Core Processor, Python 2.7.5
See https://gist.github.com/043333e198562245b66a14b294a5023a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2066.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/1062a4d50ea6492be8493b58799ad95c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2419.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9ede0f196b1268ff6c4d48c9005affcb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3171.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/433d28f5d56da133ec030c15dd04001a for a full test report.

File: easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-intel-2018b-ASE-3.18.0-Python-3.6.6.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+]
+checksums = [
+    '79dee367d695d68409c4d69edcbad5c8679137d6715da403f6c2500cb2178c2a',  # gpaw-19.8.1.tar.gz
+    'fd3668ae52683d4e36e9c8b4e3e69faa93d949601d1128662dc31be61876610c',  # GPAW-19.8.1-customize-intel.patch
+]
+
+dependencies = [
+    ('Python', '3.6.6'),
+    ('ASE', _aseversion, _pythonsuffix),
+    ('libxc', '3.0.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-intel-2018b-ASE-3.18.0-Python-3.6.6.eb]: @schiotz Existing easyconfigs for `GPAW` 19.8.1 were using `libxc` 4.3.4.

Is there any reason you're not using `libxc` 4.3.4 here, but 3.0.1 instead?

Comment (boegel) [reply easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-intel-2018b-ASE-3.18.0-Python-3.6.6.eb]: Note that the existing easyconfigs for `libxc` 4.2.3 that use a `2018b` toolchain would do?

Comment (schiotz) [reply easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-intel-2018b-ASE-3.18.0-Python-3.6.6.eb]: @boegel 
The only reason for using libxc 3.0.1 is that some other software uses that version, so a conflict occurs if I switch to 4.2.3.  Other than that, 4.2.3 would be preferable.

Comment (boegel) [reply easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-intel-2018b-ASE-3.18.0-Python-3.6.6.eb]: You mean `AtomPAW` and `ABINIT`?

Maybe it's good to mention that in a comment above this line?

Comment (schiotz) [reply easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-intel-2018b-ASE-3.18.0-Python-3.6.6.eb]: Yes, that's the ones.  I will add a comment.  In the 2019a toolchains we use a newer libxc.

File: easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-foss-2018b-ASE-3.18.0-Python-3.6.6.eb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+easyblock = "PythonPackage"
+
+name = 'GPAW'
+version = '19.8.1'
+pythonsuffix = '-Python-%(pyver)s'
+aseversion = '3.18.0'
```
Comment (boegel) [inline easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-foss-2018b-ASE-3.18.0-Python-3.6.6.eb]: @schiotz You'll need to rename these local variables to `local_pythonsuffix` and `local_aseversion`, see https://easybuild.readthedocs.io/en/develop/Easyconfig-files-local-variables.html for more information

Comment (boegel) [reply easybuild/easyconfigs/g/GPAW/GPAW-19.8.1-foss-2018b-ASE-3.18.0-Python-3.6.6.eb]: took the liberty to fix this myself in 985f5bae78, so we can get this merged...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3121.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/b624322cf2a802298e75293c770239d8 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @schiotz!

**Comment by schiotz** [comment]
@boegel Thank you very much indeed for this.

**Pull request merged**